clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationForm.tsx

```diff
@@ -0,0 +1,408 @@
+  allDataUses: DataUse[];
+  allDataSubjects: DataSubject[];
+  allDatasets?: Dataset[];
+}
+
+export const PrivacyDeclarationFormComponents = ({
+  allDataUses,
+  allDataCategories,
+  allDataSubjects,
+  allDatasets,
+  values,
+  privacyDeclarationId,
+}: DataProps & {
+  values: FormValues;
+  privacyDeclarationId?: string;
+}) => {
+  const legalBasisForProcessingOptions = (
+    Object.keys(LegalBasisForProcessingEnum) as Array<
+      keyof typeof LegalBasisForProcessingEnum
+    >
+  ).map((key) => ({
+    value: LegalBasisForProcessingEnum[key],
+    label: LegalBasisForProcessingEnum[key],
+  }));
+
+  const legalBasisForSpecialCategoryOptions = (
+    Object.keys(SpecialCategoryLegalBasisEnum) as Array<
+      keyof typeof SpecialCategoryLegalBasisEnum
+    >
+  ).map((key) => ({
+    value: SpecialCategoryLegalBasisEnum[key],
+    label: SpecialCategoryLegalBasisEnum[key],
+  }));
+
+  const datasetSelectOptions = allDatasets
+    ? allDatasets.map((ds) => ({
+        value: ds.fides_key,
+        label: ds.name ? ds.name : ds.fides_key,
+      }))
+    : [];
+
+  return (
+    <Stack spacing={4}>
+      <SystemFormInputGroup heading="Data use declaration">
+        <CustomTextInput
+          id="name"
+          label="Declaration name (optional)"
+          name="name"
+          tooltip="Would you like to append anything to the system name?"
+          variant="stacked"
+        />
+        <CustomSelect
+          id="data_use"
+          label="Data use"
+          name="data_use"
+          options={allDataUses.map((data) => ({
+            value: data.fides_key,
+            label: data.fides_key,
+          }))}
+          tooltip="For which business purposes is this data processed?"
+          variant="stacked"
+          isRequired
+          isDisabled={!!privacyDeclarationId}
+        />
+        <CustomSelect
+          name="data_categories"
+          label="Data categories"
+          options={allDataCategories.map((data) => ({
+            value: data.fides_key,
+            label: data.fides_key,
+          }))}
+          tooltip="Which categories of personal data are collected for this purpose?"
+          isMulti
+          isRequired
+          variant="stacked"
+        />
+        <CustomSelect
+          name="data_subjects"
+          label="Data subjects"
+          options={allDataSubjects.map((data) => ({
+            value: data.fides_key,
+            label: data.fides_key,
+          }))}
+          tooltip="Who are the subjects for this personal data?"
+          isMulti
+          variant="stacked"
+        />
+
+        <CustomSelect
+          name="data_sources"
+          label="Data sources"
+          options={[]}
+          tooltip="Where do these categories of data come from?"
+          isMulti
+          variant="stacked"
+        />
+        <Stack spacing={0}>
+          <CustomSelect
+            name="legal_basis_for_processing"
+            label="Legal basis for processing"
```

Comment:
I'm running into some issues with this field not saving

Reply:
I can't reproduce this, what's the issue you're having?